build/common.sh

```diff
@@ -950,6 +950,20 @@ function kube::release::package_kube_manifests_tarball() {
   mkdir -p "${dst_dir}/dns"
   tar c -C "${KUBE_ROOT}/cluster/saltbase/salt/kube-dns" ${objects} | tar x -C "${dst_dir}/dns"
 
+  # We leave the `{{ pillar['federations_domain_map'] }}` parameter as is, if
+  # the right federation environment variables isn't set. This is to allow
+  # users to provide these pillar values using the regular salt's mechanisms
+  # during cluster bootstrap.
+  if [[ "${FEDERATION:-}" == "true" ]]; then
+    FEDERATIONS_DOMAIN_MAP="${FEDERATIONS_DOMAIN_MAP:-}"
+    if [[ -z "${FEDERATIONS_DOMAIN_MAP}" && -n "${FEDERATION_NAME:-}" && -n "${DNS_ZONE_NAME:-}" ]]; then
```

Comment:
Yes as discussed, we should do one: default to a reasonable value or throw an error.
Looks like you dont like defaulting so we can throw an error :)

Dont want to block this PR. Lets get this PR merged to get the tests passing.
You can send a follow up PR.